src/odemis/gui/comp/overlay/ellipse.py

```diff
@@ -97,47 +97,54 @@ def restore_state(self, state: EllipseState):
         self._phys_to_view()
         self.points.value = self._points
 
-    def is_point_in_shape(self, v_point):
+    def check_point_proximity(self, v_point):
+        """
+        Determine if the view point is in the proximity of the shape.
+
+        Proximity is defined as either:
+        - Near the edges of the shape.
+        - Inside the shape itself.
+
+        :param: v_point: The point in view coordinates.
+        :returns: whether the view point is near the edges or inside the shape.
+        """
         # Use rectangle points instead of circumference points to make the calculation faster
         # Also editing the ellipse edges is the same as editing the rectangle's edges
         rectangle_points = self.get_physical_sel()
         if rectangle_points:
             hover, _ = self.get_hover(v_point)
-            return bool(hover)
+            return hover != gui.HOVER_NONE
         return False
 
     def on_left_up(self, evt):
         """
         Check if left click was in ellipse. If so, activate the overlay. Otherwise, deactivate.
         """
-        if self.active.value:
-            # If the Diagonal points are not the same means the rectangle has been created
-            if self.p_point1 != self.p_point3:
-                # Activate/deactivate region
-                self._view_to_phys()
-                rectangle_points = self.get_physical_sel()
-                if rectangle_points:
-                    self.selected.value = self.is_point_in_shape(evt.Position)
-                    if self.selected.value:
-                        self.set_physical_sel(rectangle_points)
-
-            # SelectionMixin._on_left_up has some functionality which does not work here, so only call the parts
-            # that we need
-            self.clear_drag()
-            self.selection_mode = SEL_MODE_NONE
-            self.edit_hover = None
-
-            # Set the points VA after drawing because draw() gathers the points
-            self.cnvs.update_drawing()
-            if self.selected.value:
-                self.points.value = self._points
-        WorldOverlay.on_left_up(self, evt)
+        # If the Diagonal points are not the same means the rectangle has been created
```

Comment:
This code is for ellipses, is "rectangle" a copy paste error? Otherwise I don't think I understand the comment :)

Same for `rectangle_points` on line 127 and 128

Reply:
Well, this is correct. We get the rectangle points 1, 2, 3, 4 first. Then the ellipse points are drawn inside the rectangle in the `draw()` method. You see this happening in line 140 where an ellipse circumference points are drawn based on rectangles points 1, 2, 3, 4 and finally the `point VA` value is assigned